crates/sui-kvstore/src/writer.rs

```diff
@@ -137,29 +137,48 @@ where
                 .get_full_checkpoint_contents(&checkpoint_summary.content_digest)
                 .map_err(|_| anyhow!("Failed to read checkpoint content from store"))?
             {
-                let mut transactions = vec![];
-                let mut effects = vec![];
-                let mut events = vec![];
-
-                for content in contents.iter() {
-                    let transaction_digest = content.transaction.digest().into_inner().to_vec();
-                    effects.push((transaction_digest.clone(), content.effects.clone()));
-                    transactions.push((transaction_digest, content.transaction.clone()));
-
-                    if let Some(event_digest) = content.effects.events_digest() {
-                        if let Some(tx_events) = store
-                            .get_transaction_events(event_digest)
-                            .map_err(|_| anyhow!("Failed to fetch events from the store"))?
-                        {
-                            events.push((event_digest.into_inner().to_vec(), tx_events));
-                        }
-                    }
-                }
+                // let mut transactions = vec![];
+                // let mut effects = vec![];
+                // let mut events = vec![];
+                //
+                // for content in contents.iter() {
+                //     let transaction_digest = content.transaction.digest().into_inner().to_vec();
+                //     effects.push((transaction_digest.clone(), content.effects.clone()));
+                //     transactions.push((transaction_digest, content.transaction.clone()));
+                //
+                //     if let Some(event_digest) = content.effects.events_digest() {
+                //         if let Some(tx_events) = store
+                //             .get_transaction_events(event_digest)
+                //             .map_err(|_| anyhow!("Failed to fetch events from the store"))?
+                //         {
+                //             events.push((event_digest.into_inner().to_vec(), tx_events));
+                //         }
+                //     }
+                // }
+                // client
+                //     .multi_set(KVTable::Transactions, transactions)
+                //     .await?;
+                // client.multi_set(KVTable::Effects, effects).await?;
+                // client.multi_set(KVTable::Events, events).await?;
+                let checkpoint_keys = [
+                    checkpoint_summary.sequence_number.to_be_bytes().to_vec(),
```

Comment:
the sequence number should be encoded with https://github.com/MystenLabs/sui/blob/main/crates/sui-storage/src/http_key_value_store.rs#L38-L46